sdk/core/azure-core/src/main/java/com/azure/core/http/policy/HttpLoggingPolicy.java

```diff
@@ -309,9 +309,12 @@ private ClientLogger getHttpLoggingPolicyLogger() {
      * @return A URL with query parameters redacted based on configurations in this policy.
      */
     private static String getRedactedUrl(URL url, Set<String> allowedQueryParameterNames) {
-        return UrlBuilder.parse(url)
-            .setQuery(getAllowedQueryString(url.getQuery(), allowedQueryParameterNames))
-            .toString();
+        UrlBuilder builder = UrlBuilder.parse(url);
+        String allowedQueryString = getAllowedQueryString(url.getQuery(), allowedQueryParameterNames);
+        return builder
+                .clearQuery()
```

Comment:
For my own understanding, why is `clearQuery()` required here? It would be nice to have a // comment here to clarify why this is required.

Reply:
1. setQuery() does not replace the full query map in UrlBuilder, as it probably should. This is legacy behavior. Our first idea was to implement clearQuery() (which just clears the associated name/value `Map`) in setQuery() to give it full replacement behavior. Storage SDK however uses setQuery() as an appending mechanism:

`setQuery("name=value")`
`setQuery("name2=value2")`
`String s = builder.toString();`

`// s = name=value&name2=value2 instead of only "name2=value2".`

2. `clearQuery()` therefore needs to be called from client code , to explicitly clear `UrlBuilder.query` map and start building the query string from scratch.

Comment added to the code.